### PR TITLE
Fix ses smtp accesskey

### LINKED ISSUE #717: iam accesskey smtp password generation fails
### What happened?
smtp password is region specific so the region has to be specified in the provider block to be generated correctly by the TF see ->
https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/iam/access_key.go#L125

this conflics with 
-> 
https://github.com/upbound/provider-aws/blob/main/config/overrides.go#L20

### How can we reproduce it?
try and use the smtp PWD generated for an ses server / accesskey in ap-southeast-1 region

### What environment did it happen in?

* Crossplane Version: 12
* Provider Version: 0.34.0

### PR DESCRIPTION
**Pull request by alexinthesky**
### Description of your changes

re-add region to aws_iam_access_key
Fixes #717

I have:

- [X] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

**Review by turkenh — COMMENTED** [review]
Thank you for your contribution @alexinthesky 🙌

File: config/overrides.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 // does not have a region notion.
 func RegionAddition() config.ResourceOption {
 	return func(r *config.Resource) {
```
[the remark this answers is not in the document]
Comment (turkenf) [reply config/overrides.go:23]: Fixed.

File: config/overrides.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
...
-		if r.ShortGroup == "iam" || r.ShortGroup == "opsworks" {
+		if (r.ShortGroup == "iam" || r.ShortGroup == "opsworks") && r.Name != "aws_iam_access_key" {
```
[the remark this answers is not in the document]
Comment (turkenf) [reply config/overrides.go]: Done.

**Comment by turkenf** [comment]
/test-examples="examples/iam/user.yaml"

**Review by turkenh — CHANGES_REQUESTED** [review]
Hmm, this would indeed be a breaking change for IAM AccessKey resource users since their existing resource will start failing with the lack of `region` parameter.

I think we would also need to add it as optional instead of required.

**Comment by alexinthesky** [comment]
yeah this is even breaking everything in iam group

**Comment by turkenf** [comment]
Hi, @alexinthesky are you still interested in moving this PR forward?

**Comment by turkenf** [comment]
@alexinthesky, I'm closing this PR for now, if you're still interested and want to move it forward, feel free to reopen it.

**Comment by tdjordan** [comment]
I am running into a situation where I want to create an AccessKey for a User but get the `ses_smtp_password_v4` for a different region than what is generated by default.

For example, the `ses_smtp_password_v4` for `us-west-2` instead of the currently generated one for `us-east-2`.

I am interested in re-opening this PR to allow the region to be specified for the `aws_iam_access_key` resource.

What is needed for me to do to help drive this PR into an acceptable merge state?

I have little experience in `go` but am willing to learn and facilitate.

Alternatively, I have looked at the surrounding code and it is currently unclear to me what is populating the Region in the AWSCLient context. Could someone point me in the right direction?

**Pull request closed without merging**